rpcs3/rpcs3qt/gs_frame.cpp

```diff
@@ -201,14 +209,33 @@ void gs_frame::toggle_fullscreen()
 		if (visibility() == FullScreen)
 		{
 			setVisibility(Windowed);
+			// in windowed mode we default to not hiding / locking the mouse
+			m_mouse_hide_and_lock = false;
 		}
 		else
 		{
 			setVisibility(FullScreen);
+			// in fullscreen (unless we want to show mouse) then we default to hiding and locking
+			if (!m_show_mouse_in_fullscreen)
+				m_mouse_hide_and_lock = true;
```

Comment:
So you want to keep the mouse locked everytime we switch from full screen to windowed?
I can imagine that this is somewhat unwanted, since the lock should be defaulted to off as there is no lock in current master anyway.

In that case wouldn't it be better to say:
`m_mouse_hide_and_lock = !m_show_mouse_in_fullscreen;`

Reply:
?

If I read things right he added a toggle that makes the locking behaviour default to "uncaptured" (aka. `Show mouse cursor in Fullscreen`) so that multi-monitor people who don't have their mice bound can move their cursor out of the window. The lines quoted just act that out, I'm not sure why you're saying that alters the fs->w behaviour (esp since I tested the build with this inside and it works the same as master):

Now it's a good question to think about what happens if someone has multiple displays and does use mouse binding, whether it would be better for them to default to locking or not (I'd assume yes), but I'm not sure there's code existing that checks if mouse events are mapped to anything, so that'd need adding then and this condition expanded upon.